### ISSUE TITLE
Not all getter/setter references are detected when using JavaScript 

### ISSUE THREAD

**Issue by jespertheend** [microsoft/TypeScript#49822]
# Bug Report

### 🕗 Version & Regression Information

```ts
export class Foo {
	constructor() {
        this.bar = false;
		this.bar = true;
	}
    
	get bar() {
		return true
	}

	set bar(value) {
		// Do something with value
	}
}

/** @type {Foo[]} */
const items = [];

for (const item of items) {
    item.bar = true;
}
```

### 🙁 Actual behavior

When clicking `this.bar` on line 3, only the instances on line 3 and 4 are highlighted. When pressing `F2` only those two instances are renamed, and when pressing `F12` (cmd `F12` on the playground) only those two instances appear in the list of references.

### 🙂 Expected behavior

All references, including the getter, setter and `item.bar` on line 20 are referenced. This works fine when using TypeScript. See [this playground example](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwDYEMDOa4DEITgbwCgBIBCAOzRigFcEZoAKASgMLg845gAsBLNADoARiihwAvHABmKJGmABuEsV4CRYyXGo0lJAL4kSAc2DxRUFm2LEoZmlHLbawA4RILzYxgDc5u1iIbAHpguAARPDQIAFszfnJjOAB3Pl44PyRdN0NCMkp4NOAYzAAubFwAbQBdLRrlQmloOEZ8qjgimLgIaQ6YYrRA9k5OjXEpHT19IA) for the TypeScript equivalent of this code.

https://user-images.githubusercontent.com/2737650/177770729-e9c893c6-7c8a-4228-8fb6-865256cec662.mp4

**Comment by komyg** [comment]
Hi @RyanCavanaugh, I would like to help on this issue. Could you assign it to me?

**Comment by komyg** [comment]
Oh, now I see. So is this a problem when using the Typescript language server to process Javascript files?

**Comment by jespertheend** [comment]
Yeah, if you use `checkJs: true` in your `tsconfig.json`/`jsconfig.json` you can use JavaScript files almost the same way you use TypeScript. The only difference is that you have to use JSDoc syntax to assign types rather than being able to do `const x : number = 1`. Though I don't think that flag is required in order to reproduce this in the language server.

**Comment by komyg** [comment]
I've tested it, and you don't need the flag.

Having a _.js_ file is enough to reproduce the error.

**Comment by komyg** [comment]
@RyanCavanaugh I'd like to help out with this bug. Could you assign it to me?

**Comment by RyanCavanaugh** [comment]
@komyg glad to hear! It's not possible for me to assign this (you'd need a permission level on the repo that we don't issue to non-Microsoft employees). See also https://github.com/microsoft/TypeScript/blob/main/CONTRIBUTING.md#issue-claiming

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
